budgetportal/openspending.py

```diff
@@ -121,6 +121,17 @@ def aggregate(self, cuts=None, drilldowns=None):
             raise Exception("More cells than expected - perhaps we should start paging")
         return aggregate_result.json()
 
+    def filter_dept(self, result, dept_name, sphere, financial_year):
+        sphere_filter = {
+            'national': 'vote_number.department',
+            'provincial': 'department.department'
+        }
+        filtered_results = []
+        for budget in result['cells']:
+            if budget[sphere_filter[sphere]] == dept_name and budget['financial_year.financial_year'] == financial_year:
```

Comment:
here we should also be using self.get_financial_year_ref() or whatever it is instead of 'financia_year.financial_year'